### ISSUE TITLE
Sandbox Login - Authentication Failure

### ISSUE THREAD

**Issue by JPWallhorn** [restforce/restforce#335]
Hey guys,

I went through everything but for some reason none if the instructions / help work for me. I'm still getting an authentication error when I try to login to a sandbox. 

This is my setup:
client = Restforce.new :refresh_token => current_user.refresh_token,
        :instance_url  => current_user.instance_url,
        :client_id     => 'client_id',
        :client_secret => 'client_secret',
        :authentication_callback => Proc.new { |x| Rails.logger.debug x.to_s },
        :api_version => '38.0',
        :host => 'test.salesforce.com'

I tried using our custom salesforce domain instead of test.salesforce.com but nothing works. Still getting an authentication error.

For the production environment it works just fine. I hope you guys can help me. 

Thank you in advance

**Comment by timrogers** [comment]
I'd try talking to Salesforce about this to understand better what settings you should be using - and maybe look at https://github.com/restforce/restforce/issues/284.

**Comment by antwonlee** [comment]
@JPWallhorn have you tried with user and password method to connect to your sandbox?

**Comment by JPWallhorn** [comment]
No, I'm trying to avoid username / password. 

I got one step further but looking at OmniAuth. I added `provider OmniAuth::Strategies::SalesforceSandbox,` 

Additionally, when I want to login to Salesforce now I get directed to test.salesforce.com rather than login.salesforce.com so it chooses the right provider now. 

Now I'm just running into an issue that I have an URI mismatch: `error=redirect_uri_mismatch&error_description=redirect_uri%20must%20match%20configuration`
